tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -787,19 +787,16 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::Internal::RNTupleDescripto
       if (columnCardinality == 0)
          continue;
 
-      // We already know (from AddColumn()) that the column indexes and representation indexes are consecutive
+      // In AddColumn, we already checked that all but the last representation are complete.
+      // Check that the last column representation is complete, i.e. has all columns.
       const auto &logicalColumnIds = fieldDesc.GetLogicalColumnIds();
       const auto nColumns = logicalColumnIds.size();
-      if ((nColumns == 0) || ((nColumns % columnCardinality) != 0)) {
-         return R__FAIL("field with id '" + std::to_string(fieldId) + "' has an invalid number of columns");
-      }
+      if (nColumns <= columnCardinality)
+         continue;
```

Comment:
Isn't this an error as well because it means the field doesn't have all columns? On the other hand, the previous condition allowed `nColumns == 0`, but shouldn't this already be taken care of by `columnCardinality == 0`?

Reply:
The condition triggers if we have only a single column representation. I this case, by construction we have `nColumns + 1 == columnCardinality`. Perhaps not the clearest way to express that in the current `if`